lib/node-package-manager.ts

```diff
@@ -1,46 +1,70 @@
 import * as path from "path";
+import { exported } from "./common/decorators";
 
 export class NodePackageManager implements INodePackageManager {
 	constructor(private $fs: IFileSystem,
 		private $hostInfo: IHostInfo,
 		private $errors: IErrors,
 		private $childProcess: IChildProcess,
-		private $logger: ILogger,
-		private $options: IOptions) { }
+		private $logger: ILogger) { }
 
-	public async install(packageName: string, pathToSave: string, config?: any): Promise<any> {
-		if (this.$options.disableNpmInstall) {
+	@exported("npm")
+	public async install(packageName: string, pathToSave: string, config: INodePackageManagerInstallOptions): Promise<INpmInstallResultInfo> {
+		if (config.disableNpmInstall) {
 			return;
 		}
-		if (this.$options.ignoreScripts) {
-			config = config || {};
+		if (config.ignoreScripts) {
 			config["ignore-scripts"] = true;
 		}
 
 		let packageJsonPath = path.join(pathToSave, "package.json");
 		let jsonContentBefore = this.$fs.readJson(packageJsonPath);
-		let dependenciesBefore = _.keys(jsonContentBefore.dependencies).concat(_.keys(jsonContentBefore.devDependencies));
 
 		let flags = this.getFlagsString(config, true);
 		let params = ["install"];
-		if (packageName !== pathToSave) {
-			params.push(packageName); //because npm install ${pwd} on mac tries to install itself as a dependency (windows and linux have no such issues)
+		const isInstallingAllDependencies = packageName === pathToSave;
```

Comment:
When packageName is pathToSave ? When packageName is full path to `.tgz` maybe ?